src/subcommand/init.rs

```diff
@@ -0,0 +1,104 @@
+//! Implementation for the `init` subcommand.
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use crate::ext::{
+    error_stack::{ErrorHelper, IntoContext},
+    result::WrapOk,
+};
+use error_stack::Result;
+use indoc::formatdoc;
+
+/// Errors encountered during init.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// A config file already exists
+    #[error("config file exists")]
+    ConfigFileExists,
+
+    /// Creating the data root directory
+    #[error("create data root")]
+    CreateDataRoot(PathBuf),
+
+    /// Writing the file did not work
+    #[error("write config file")]
+    WriteConfigFile(PathBuf),
+}
+
+/// generate the config and db files in the default location
+#[tracing::instrument(skip_all)]
+pub fn main(data_root: &PathBuf) -> Result<(), Error> {
+    let default_already_exists = write_config(data_root, "config.yml", false)?;
+    write_config(data_root, "config.example.yml", true)?;
+    if default_already_exists {
+        let output = formatdoc! {r#"
+
+        `broker init` detected a previously existing config file at {config} and left it as is.
+
+        `broker init` did, however, create a new example config file for you at {example_config}.
+
+        This example config file contains a detailed explanation of everything you need to do to get broker up and running.
+
+        You can safely re-run `broker init` at any time to re-generate the "config.example.yml" file.
+        "#,
+            config = data_root.join("config.yml").display(),
+            example_config = data_root.join("config.example.yml").display(),
+        };
+        println!("{}", output);
+    } else {
+        let output = formatdoc! {r#"
+
+        `broker init` created an example config in {config}.
+
+        The config file contains a detailed explanation of everything you need to do to get broker up and running.
+
+        The next step is to open {config} and follow the instructions to configure broker.
+
+        We also wrote the same example file to {example_config} to serve as a reference for you in the future.
+
+        You can safely re-run `broker init` at any time to re-generate the "config.example.yml" file.
+
+        "#,
+            config = data_root.join("config.yml").display(),
+            example_config = data_root.join("config.example.yml").display(),
+        };
+        println!("{}", output);
+    }
+    Ok(())
+}
+
+fn write_config(data_root: &PathBuf, filename: &str, force_write: bool) -> Result<bool, Error> {
+    let config_file_path = data_root.join(filename);
+    if config_file_path.try_exists().unwrap_or(false) && !force_write {
+        return true.wrap_ok();
```

Comment:
For values like this I'd just use `Ok(true)`. The intent with `wrap_ok` is to reduce the need to double up parens: `Ok(some_function(some_value.some_method()))` reads less cleanly than `some_function(some_value.some_method()).wrap_ok()`, and if you have a chain of operations it's way worse.

But `true.wrap_ok()` reads less cleanly than just `Ok(true)`, so I recommend against using it in simple scenarios like that.